assets/js/main.js

```diff
@@ -11,7 +11,94 @@ const giveConsentButton  = document.querySelector( '.give-consent' ),
 	cookiePrefsButton    = document.querySelector( '.view-preferences' ),
 	applyCookiePrefs     = document.querySelector( '.apply-cookie-preferences' ),
 	cookieUpdatedMessage = document.querySelector( '.consent-updated-message' ).classList,
-	closeUpdatedMessage  = document.getElementById( 'consent-close-updated-message' );
+	closeUpdatedMessage  = document.getElementById( 'consent-close-updated-message' ),
+	consentBanner        = document.getElementById( 'cookie-consent-banner' );
+
+/**
+ * Check if a user has given consent for a specific category.
+ *
+ * Wrapper function for wp_has_consent.
+ *
+ * @param {string} category The category to check consent against.
+ */
+function hasConsent( category ) {
```

Comment:
Only just realised as well these functions are added to the global scope - it's important to consider an environment where there might be 1st party JS and other things adding stuff to the global scope. A function called `getCookie()` or `setCookie()` could likely already exist or be easily / accidentally overwritten.

We have a general pattern similar to WordPress namespacing everything under a `wp` object eg. `wp.data` for Altis. So the code would become something like:

```js
var Altis = window.Altis || {};
Altis.Consent = {};

Altis.Consent.has = function ( category ) { 
  //...
}
```

Reply:
Beautiful. I was wondering about best practice for this.